src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
Consider adding .Start and .Run as extension to the builder. That way you could have

using (new Builder().UseEtc().Start()) {...}

Or

new Builder().UseEtc().Run();

Sent from mobile device
________________________________
From: John Luo<mailto:notifications@github.com>
Sent: ‎12/‎17/‎2015 2:16 PM
To: aspnet/Hosting<mailto:Hosting@noreply.github.com>
Cc: Louis DeJardin<mailto:lodejard@microsoft.com>
Subject: Re: [Hosting] API changes to hosting and TestServer (#521)


In src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs<https://github.com/aspnet/Hosting/pull/521#discussion_r47968381>:

> @@ -98,14 +127,14 @@ public virtual IApplication Start()
>              _applicationLifetime.NotifyStarted();
>              logger.Started();
>
> -            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
> -            {
> -                logger.Shutdown();
> -                _applicationLifetime.StopApplication();
> -                Server.Dispose();
> -                _applicationLifetime.NotifyStopped();
> -                (_applicationServices as IDisposable)?.Dispose();
> -            }));
> +            return new Disposable(() =>


        public static void Main(string[] args)
        {
            var config = WebApplicationConfiguration.GetDefault(args);

            var builder = new WebApplicationBuilder()
                .UseConfiguration(config)
                .UseStartup<StartupBlockingOnStart>();

            using (var application = builder.Build())
            {
                application.Start();
                Console.ReadLine();
            }
        }


This is what usage looks like for the alternative API where we Start() returns void. Does this look better than the current usage?

—
Reply to this email directly or view it on GitHub<https://github.com/aspnet/Hosting/pull/521/files#r47968381>.